fixes to mtlx files

HardCoreCodin opened the pull request:
Made changes to the .mtlx files and renamed some image files to make things render more properly in Maya with Arnold:
<img width="982" height="544" alt="OpenPBR_Playground_Maya_Arnold" src="https://github.com/user-attachments/assets/de1b7be1-dd15-4b86-9eda-8caa53c0f2a0" />
<img width="986" height="515" alt="OpenPBR_Playground_Maya_Arnold_after" src="https://github.com/user-attachments/assets/c3632aa2-9579-4615-a1a5-1fabe6f682fd" />

OJglass: Set base color weight to enable the graph input of the base color texture
OJglass: Set coat weight to 0 as the coat has nothing but IOR of 1
bottle: Set coat weight to 1 to enable it with its settings
bottle: Remove redundant rotation input - it is set to 1 which would have been a no-op anyway as it corresponds to 360 deg rotation
bubblesMasonJar: Set Fuzz weight to 1 to enable it and it's roughness settings
ceramicMug: Set Fuzz weight to 1 to enable it and it's roughness settings
chairA: Remove coat settings as they were not doing anything anyway
cord: Set coat weight to 1 to enable it with its settings
floorFrames: Set Fuzz weight to 1 to enable it and it's roughness settings
JellyToyEyes: Set Substuface weight to 1 to enable it and it's subsurface color settings
MasonJarGlass: Re-enabled Opacity image node and renamed it to Transmission as it holds a transmission mask
MasonJarGlass: Connect Transmission mast to transmission_weight to make use of its mask
MasonJarGlass: Increase transmission_depth by a factor of 100 as the value was too low and made the glass black
MasonJarWater: Removed coat settings as the IOR was 1
MasonJarWater: Set Subsurface weight to 1 to enable it and it's subsurface color settings
MasonJarWater: Set Subsurface radius from 5 to 0.5 to make the subsurface color map come through in the render
paper: Enable geometry_thin_walled
paperPlans: Enable geometry_thin_walled
wings: Renamed all images to remove redundant use of UDIM that causes issues in some DCCs
wings: Re-Enabled geometry_opacity connection to bring back the opacity mask

HardCoreCodin committed:
fixes to mtlx files

OJglass: Set base color weight to enable the graph input of the base color texture
OJglass: Set coat weight to 0 as the coat has nothing but IOR of 1
bottle: Set coat weight to 1 to enable it with its settings
bottle: Remove redundant rotation input - it is set to 1 which would have been a no-op anyway as it corresponds to 360 deg rotation
bubblesMasonJar: Set Fuzz weight to 1 to enable it and it's roughness settings
ceramicMug: Set Fuzz weight to 1 to enable it and it's roughness settings
chairA: Remove coat settings as they were not doing anything anyway
cord: Set coat weight to 1 to enable it with its settings
floorFrames: Set Fuzz weight to 1 to enable it and it's roughness settings
JellyToyEyes: Set Substuface weight to 1 to enable it and it's subsurface color settings
MasonJarGlass: Re-enabled Opacity image node and renamed it to Transmission as it holds a transmission mask
MasonJarGlass: Connect Transmission mast to transmission_weight to make use of its mask
MasonJarGlass: Increase transmission_depth by a factor of 100 as the value was too low and made the glass black
MasonJarWater: Removed coat settings as the IOR was 1
MasonJarWater: Set Subsurface weight to 1 to enable it and it's subsurface color settings
MasonJarWater: Set Subsurface radius from 5 to 0.5 to make the subsurface color map come through in the render
paper: Enable geometry_thin_walled
paperPlans: Enable geometry_thin_walled
wings: Renamed all images to remove redundant use of UDIM that causes issues in some DCCs
wings: Re-Enabled geometry_opacity connection to bring back the opacity mask
Signed-off-by: Arnon Marcus <hard.core.codin@gmail.com>

theblackunknown commented on ShdrPlygrnd/materials/bottle.mtlx:
```diff
@@ -65,11 +65,10 @@
     <input name="coat_ior" type="float" value="1.7" />
     <input name="coat_color" type="color3" value="0.46778, 1.00002, 0.46778" />
     <input name="coat_roughness_anisotropy" type="float" value="1" />
-    <!-- FIXME: No available on OpenPBR v1.1 -->
-    <!-- <input name="coat_rotation" type="float" value="1" /> -->
```
Don't you find those breadcrumbs comment valuable ? I keep those because I felt the original intent is not preserved with the upgrade to v1.X

HardCoreCodin replied:
The original intent is preserved. See my 4th point in the description.
Even if that input was still supported, if would still be redundant entry that does nothing.
If the value was anything else that 1 or 0, then yes, there would still be something to do to get the original intent by providing a rotated tangent to the coat tangent input (using a tangent and rotation nodes).
The FIXME comment is there to remind people that it needs looking into. Well, it's been looked into now, so there's no need to leave it there in my opinion.

theblackunknown commented on ShdrPlygrnd/materials/bubblesMasonJar.mtlx:
```diff
@@ -10,7 +10,7 @@
 
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Fuzz is not enabled by default -->
+    <input name="fuzz_weight" type="float" value="1" />
```
Have you assessed whether it makes sense to have fuzz on bubbles ?

HardCoreCodin replied:
The phrasing in the FIXME comment coupled with the explicitly provided **fuzz_roughness** non-zero value suggested that the intended fix is to turn on the layer. The explicit value provided on its own signals intent by the artist / author. Questioning that artistic intent I felt was not for me to decide. If I were to be asked, I'd probably say no, given what I know about the purpose of the fuzz layer and about bubbles.

theblackunknown replied:
Feel free to remove the fuzz setup too if you feel it is not contributing much to this look

theblackunknown commented on ShdrPlygrnd/materials/chairA.mtlx:
```diff
@@ -46,10 +46,8 @@
 
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Fuzz is not enabled by default -->
+    <input name="fuzz_weight" type="float" value="1" />
```
Have you assessed whether it makes sense to have fuzz on the chair ?

HardCoreCodin replied:
Same answer as above - personally, I'd keep it off and remove the **fuzz_rouphness** input entry here as well.

theblackunknown replied:
Agreed, feel free to remove it or keep you changes

theblackunknown commented on ShdrPlygrnd/materials/ceramicMug.mtlx:
```diff
@@ -46,7 +46,7 @@
   </normalmap>
 
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Fuzz is not enabled by default -->
```
Have you assessed whether it makes sense to have fuzz on the mug ?

HardCoreCodin replied:
Similar answer to above - is probably not really for me to decide artistically, this here again was more from a mechanical fixing perspective. May be worth trying different values for weight and roughness. Looking at reference, ceramic does tend to exhibit some sheen-like reflectance, if that was the original intent to replicate. But the fuzz layer in OpenPBR can also be used to emulate dust rather than sheen, but at higher roughness values, so it depends what is being going for artistically here.

<img width="672" height="740" alt="CeramicMug_sheen" src="https://github.com/user-attachments/assets/2c90689c-3573-45a5-a479-635557c64b02" />
<img width="684" height="880" alt="CeramicMug_FuzzRenders" src="https://github.com/user-attachments/assets/1f82322e-a42c-4d39-9d44-f17bd85f9a13" />
Note: These are renders from my implementation in Octane, the prior ones were from Arnold.

theblackunknown replied:
Sounds good, thanks for sharing renders with different combinations

theblackunknown commented on ShdrPlygrnd/materials/cord.mtlx:
```diff
@@ -1,7 +1,7 @@
 <materialx version="1.39" colorspace="lin_rec709">
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Coat is not enabled by default -->
+    <input name="coat_weight" type="float" value="1" />
```
Have you assessed whether it makes sense to have coat on the cord ?

HardCoreCodin replied:
Similar answer, there's a non-zero explicitly provided roughness value. Probably worth having, is does make it a bit more shiny, as chords tend to be. Not a huge difference in this case, but can be noticeable.
<img width="1024" height="512" alt="Cord_Coat0" src="https://github.com/user-attachments/assets/eccf6bc4-0aed-428d-973a-309ca72c02b1" />
<img width="1024" height="512" alt="Cord_Coat1" src="https://github.com/user-attachments/assets/0e4f65c0-7dfb-436f-b2b6-2b18e8b7e665" />

theblackunknown replied:
Sounds good, thanks for sharing renders with different combinations

theblackunknown commented on ShdrPlygrnd/materials/floorFrames.mtlx:
```diff
@@ -1,7 +1,7 @@
 <materialx version="1.39" colorspace="lin_rec709">
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Fuzz is not enabled by default -->
+    <input name="fuzz_weight" type="float" value="1" />
     <input name="fuzz_roughness" type="float" value="0.3" />
```
Have you assessed whether it makes sense to have fuzz on the floor frames ?

HardCoreCodin replied:
Actually, what makes even more sense is to remove the material altogether, as nothing is using it.

theblackunknown commented on ShdrPlygrnd/materials/jellyToyEyes.mtlx:
```diff
@@ -3,7 +3,7 @@
     <input name="base_color" type="color3" value="0, 0, 0" />
     <input name="specular_roughness" type="float" value="0.2" />
-    <!-- FIXME: Subsurface is not enabled by default -->
+    <input name="subsurface_weight" type="float" value="1" />
```
Have you assessed whether it makes sense to have SSS on those eyes ?

HardCoreCodin replied:
Is hard to say as it is in shadow (I've had to artificially increase exposure to see things clearly there).
Without the change the eyes are black and opaque as there's nothing to color them apart from the SSS color being set here. Also, eyes tend to have SSS, and this is a jelly toy so the rest of it's entire body is fully SSS.
Turning on SSS gives them color and depth.
<img width="768" height="512" alt="JellyToyEyes_NoSSS" src="https://github.com/user-attachments/assets/af2efdb5-6e69-40e1-8c65-8130da415d87" />
<img width="768" height="512" alt="JellyToyEyes_WithSSS" src="https://github.com/user-attachments/assets/da1116b1-58d6-4226-b4d3-0177edbc4e36" />
<img width="311" height="310" alt="JellyToyEyes_NoSSS_SpherPreview" src="https://github.com/user-attachments/assets/c9c1d9f3-e82b-4ede-9f61-12097f450e33" />
<img width="309" height="311" alt="JellyToyEyes_WithSSS_SpherPreview" src="https://github.com/user-attachments/assets/bf4fb4a8-93a7-4762-87a2-04457c36f8dd" />

theblackunknown replied:
Thanks for the extra tests, this looks like a good change

theblackunknown commented on ShdrPlygrnd/materials/masonJarGlass.mtlx:
Nice fix to the Mason Jar !

theblackunknown commented on ShdrPlygrnd/materials/masonJarWater.mtlx:
```diff
@@ -10,15 +10,13 @@
     <input name="specular_roughness" type="float" value="0.1" />
     <input name="specular_roughness_anisotropy" type="float" value="0.5" />
+    <input name="subsurface_weight" type="float" value="1" />
     <input name="subsurface_color" type="color3" nodename="ScatterColor" />
-    <input name="subsurface_radius" type="float" value="5" />
+    <input name="subsurface_radius" type="float" value="0.5" />
```
Why ? Does this make a meaningful appearance change ?

HardCoreCodin replied:
That is a tricky question to answer unfortunately:
The value is unitless, so depends on DCC unit settings.
USD has a "metersPerUnit" metadata that should be respected, ideally.
But even when units are matched, the exact curve of sensitivity of how this impacts how the SSS is evaluated, may vary across render engines. I've tested with Arnold and Octane now, they differ somewhat.
So, this change is apparent in Octane, but not in Arnold.
It controls how opaque vs. translucent the water ends up being, and its color saturation. This impacts how much of the brush in the jar is visible.
Here are some renders of Arnold vs. Octane with values of 0.05, 0.5, 5.0 and 50.0:
<img width="768" height="512" alt="Octane_MasonJarWater_SSS_radius_0_05" src="https://github.com/user-attachments/assets/9839b17f-c34e-4bc6-bb49-997fbb2ddd26" />
<img width="768" height="512" alt="Octane_MasonJarWater_SSS_radius_0_5" src="https://github.com/user-attachments/assets/57b6aded-797e-470f-a274-374a72976439" />
<img width="768" height="512" alt="Octane_MasonJarWater_SSS_radius_5_0" src="https://github.com/user-attachments/assets/b6673f9b-e384-41a5-b9c0-1fe62155dd12" />
<img width="768" height="512" alt="Octane_MasonJarWater_SSS_radius_50_0" src="https://github.com/user-attachments/assets/c2f777c6-db16-458c-b473-7b3683e4b30c" />
<img width="312" height="347" alt="Arnold_MasonJarWater_SSS_radius_0_05" src="https://github.com/user-attachments/assets/dc7056ca-76a2-48f9-8604-a31043d61f84" />
<img width="312" height="347" alt="Arnold_MasonJarWater_SSS_radius_0_5" src="https://github.com/user-attachments/assets/d6f23b72-1fa4-4a71-9bce-b9717c1958fc" />
<img width="312" height="347" alt="Arnold_MasonJarWater_SSS_radius_5_0" src="https://github.com/user-attachments/assets/5bb92c7d-200c-47b4-a40e-0211d51d2cb1" />
<img width="312" height="347" alt="Arnold_MasonJarWater_SSS_radius_50_0" src="https://github.com/user-attachments/assets/3c0ad8ae-bd6e-42c5-9f2a-8bd1e7afab93" />

theblackunknown replied:
Thanks for those extra tests !
It is the main intent is to get some murky water which you have correctly addressed.
About Arnold / Octane differences, I would not spend more time on this for now.

theblackunknown commented on ShdrPlygrnd/materials/OJglass.mtlx:
```diff
@@ -42,12 +42,8 @@
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: Why setting up a base color is the base is disabled ? -->
-    <input name="base_weight" type="float" value="0" />
+    <input name="base_weight" type="float" value="1" />
     <input name="base_color" type="color3" nodename="BaseColor" />
```
Have you noticed any meaningful appearance differences ?

HardCoreCodin replied:
The connection of the base color to a color image indicated intent - that said, it may have been a leftover from an earlier iteration where it wasn't glass. Given how it is now fully transmissive, it does nothing really, so the base color can stay 0 in this case, but then the base color connection to the image node and the image node itself should be removed.
I should also mention that this glass object was disabled for rendering in USD - whether that is intentional or not - but with a render-specific disablement. The glass does not appear in most of the renders I've seen of this content.

theblackunknown commented on ShdrPlygrnd/materials/paper.mtlx:
```diff
@@ -47,14 +47,14 @@
 
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: For paper like maeerial using thin wall is ideal ! -->
     <input name="base_color" type="color3" nodename="BaseColor" />
     <input name="specular_color" type="color3" value="0.42767, 0.42767, 0.42767" />
     <input name="specular_roughness" type="float" value="1" />
     <input name="subsurface_weight" type="float" nodename="SSSweight" />
     <input name="subsurface_color" type="color3" nodename="mtlxcolorcorrect1" />
     <input name="subsurface_radius" type="float" value="0.05" />
     <input name="geometry_normal" type="vector3" nodename="mtlxnormalmap1" />
+    <input name="geometry_thin_walled" type="boolean" value="true" />
```
Happy to see your trust :) Have you noticed any meaningful differences with this ?

HardCoreCodin replied:
You tell me :)
OpenPBR is almost 2 materials in one, in a sense. Thin Walled mode is very different ang give completely different meaning and purpose for the same inputs, especially for the subsurface slab which is connected to upstream nodes here.
<img width="1024" height="512" alt="PaperThinWalledOff" src="https://github.com/user-attachments/assets/3cdb1694-498f-4cf9-af66-4ee4d4db936e" />
<img width="1024" height="512" alt="PaperThinWalledOn" src="https://github.com/user-attachments/assets/1c1f5ead-aad9-4631-9a8b-275d2d3ff064" />

theblackunknown replied:
Agreed on all your observations, to me the thin walled makes much more sense, and yes because of the Openpbr "2 in 1" material :)
Thanks for sharing renders with different combinations

theblackunknown commented on ShdrPlygrnd/materials/paperPlans.mtlx:
```diff
@@ -15,14 +15,14 @@
 
   <open_pbr_surface name="mtlxopen_pbr_surface" type="surfaceshader">
-    <!-- FIXME: For paper like material using thin wall is ideal ! -->
     <input name="base_color" type="color3" nodename="BaseColor" />
     <input name="specular_color" type="color3" value="0.4129, 0.4129, 0.4129" />
     <input name="specular_roughness" type="float" value="0.4" />
     <input name="subsurface_weight" type="float" value="0.225" />
     <input name="subsurface_color" type="color3" nodename="mtlxcolorcorrect1" />
     <input name="subsurface_radius" type="float" value="0.2" />
     <input name="subsurface_radius_scale" type="color3" value="1, 1, 1" />
+    <input name="geometry_thin_walled" type="boolean" value="true" />
```
Happy to see your trust :) Have you noticed any meaningful differences with this ?

HardCoreCodin replied:
See above comment (yes).
<img width="1024" height="512" alt="PaperPlansThinWalledOff" src="https://github.com/user-attachments/assets/a55c9b07-913c-43a7-86db-823bf61e97a2" />
<img width="1024" height="512" alt="PaperPlansThinWalledOn" src="https://github.com/user-attachments/assets/c452cf8c-ff7c-4fb2-a290-3537a02fc451" />

theblackunknown commented on ShdrPlygrnd/materials/wings.mtlx:
```diff
@@ -5,23 +5,23 @@
+      <input name="file" type="filename" colorspace="srgb_tx" value="textures/wings_MAT_BaseColor.png" />
       <input name="texcoord" type="vector2" nodename="st" />
   </image>
   <image name="Metalness" type="float">
-      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Metalness.<UDIM>.png" />
+      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Metalness.png" />
       <input name="texcoord" type="vector2" nodename="st" />
   </image>
   <image name="Roughness" type="float">
-      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Roughness.<UDIM>.png" />
+      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Roughness.png" />
       <input name="texcoord" type="vector2" nodename="st" />
   </image>
   <image name="Opacity" type="float">
-      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Opacity.<UDIM>.png" />
+      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Opacity.png" />
       <input name="texcoord" type="vector2" nodename="st" />
   </image>
   <image name="Normal" type="vector3">
-      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Normal.<UDIM>.png" />
+      <input name="file" type="filename" colorspace="Raw" value="textures/wings_MAT_Normal.png" />
       <input name="texcoord" type="vector2" nodename="st" />
```
Nice !

theblackunknown commented on ShdrPlygrnd/materials/wings.mtlx:
```diff
@@ -53,8 +53,7 @@
     <input name="specular_roughness" type="float" nodename="Roughness" />
-    <!-- FIXME: Issue with Opacity UDIM in Arnold>
-    <input name="geometry_opacity" type="float" nodename="Opacity" /-->
+    <input name="geometry_opacity" type="float" nodename="Opacity" />
```
Nice !

theblackunknown reviewed (commented):
Thank your for those improvements @HardCoreCodin !
I have left a few comments to challenge some changes to make sure they bring visual improvements, if not I would prefer we revert those and assess first.

theblackunknown reviewed (approved):
Looks good to me !
Thanks for your patience with my comparison requests, the renders you shared looks great !
I guess for the remaining ones I'll make a leap of faith as they do seem useful and address the `TODO` / `FIXME` I had left in there.

Approved, but you still need to sort out the CLA part to be able to merge it

HardCoreCodin committed:
bubbles: Disabled fuzz
chairA: Disabled fuzz
floorFrames: Removed (unused)
OJglass: Removed base color image

Signed-off-by: Arnon Marcus <hard.core.codin@gmail.com>

molssongroup commented:
/easycla

matthewlow-dwa commented:
Thank you so much for this contribution @HardCoreCodin !  Some really nice improvements. And glad to see the CLA has now been resolved.

It appears there are conflicts and that the branch cannot be merged -- could you rebase your changes onto the latest `main`? It seems some stray commits are in the PR. Once updated, I can approve and get this merged.

matthewlow-dwa reviewed (approved):
LGTM!
